src/components/Plugin/SafeSnap/CustomBlock.vue

```diff
@@ -0,0 +1,307 @@
+<template>
+  <Block
+    title="SafeSnap"
+    :icon="!loading && moduleAddress ? 'refresh' : undefined"
```

Comment:
This block look empty for few second, you could add this param: 
```
:loading="loading"
```
This will render a loading indicator inside the block until content is loaded.